custom/fitness-demo/contexts/ClassStateProvider.js

```diff
@@ -0,0 +1,48 @@
+import React, {
+  createContext,
+  useContext,
+  useCallback,
+  useMemo,
+  useEffect,
+} from 'react';
+import { useUIState, VIEW_MODE_SPEAKER, VIEW_MODE_GRID } from '@custom/shared/contexts/UIStateProvider';
+import { useSharedState } from '@custom/shared/hooks/useSharedState';
+import PropTypes from 'prop-types';
+
+export const PRE_CLASS_LOBBY = 'Pre-class lobby';
```

Comment:
One thing that feels confusing to me from a UX perspective is the pre-class, class, post-class UI. They seem to all be the same so it's hard to tell which one you're in. (E,g. if you are screen sharing and end the class, the screen share is in the post-class too). Not necessarily any action to be taken now but it can be hard to get the full demo effect when they stages of a class feel a little conflated.

That being said!!! It's really nice to have a demo with the 3 stages. This comment is more so about thinking about the UX details of it.

Reply:
True I agree with this, @jptaylor shall we stop screen share once they end the class, what do you think about this?